lib/puppet/pops/validation/tasks_checker.rb

```diff
@@ -71,12 +71,12 @@ def check_SiteDefinition(o)
 
   def check_ApplyExpression(o)
     if in_ApplyExpression?
-      acceptor.accept(Issues::TASK_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, o, {:operation => o.class.to_s})
+      acceptor.accept(Issues::TASK_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, o, {:operation => @label_provider.label(o)})
     end
   end
 
   def illegalTasksExpression(o)
-    acceptor.accept(Issues::CATALOG_OPERATION_NOT_SUPPORTED_WHEN_SCRIPTING, o, {:operation => o.class.to_s})
+    acceptor.accept(Issues::CATALOG_OPERATION_NOT_SUPPORTED_WHEN_SCRIPTING, o, {:operation => @label_provider.label(o)})
```

Comment:
The issue objects have a mechanism to interpolate the label and handle a/an - see `issues.rb` and look at how it uses things like `label.a_an(klass)` - the effect is the same as doing it when creating the issue (like you do here), but seems nicer that the formatting is decided completely by by the issue logic - i.e. just give it the klass of the object, or the instance (depending on what you want) and then deal with a/an/the and the human readable label in the message. Naturally, there must be entries in the label provider for the apply feature AST classes - that is where you would do some mapping to for example "manifest block".